clang/SourceRange.d

```diff
@@ -0,0 +1,45 @@
+/**
+ * License: $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost Software License 1.0)
+ */
+
+module clang.SourceRange;
+import std.conv;
+import clang.c.Index;
+import clang.SourceLocation;
+
+struct SourceRange
+{
+    CXSourceRange _impl;
+
+    @property
+    SourceLocation start() const
+    {
+        return SourceLocation(clang_getRangeStart(_impl));
+    }
+
+    @property
+    SourceLocation end() const
+    {
+        return SourceLocation(clang_getRangeEnd(_impl));
+    }
+
+    @property
+    string path() const 
+    {
+        return start.path;
+    }
+
+    @property
+    string toString() const 
+    {
+        import std.format: format;
+        return format("SourceRange(%s, %s)", to!string(start), to!string(end));
```

Comment:
"Please include the field names"
Currently it looks like "SourceRange(SourceLocation(file,116,9,3735), SourceLocation(file,116,38,3764))" - isn't it good enough?

"format will convert all its arguments to string, that's the whole point :wink:. No need to call to!string"
Good point.